### ISSUE TITLE
[device-port

### ISSUE THREAD

**Issue by Qiangong2** [Halium/projectmanagement#62]
Tree: Halium-7.1

- [x] Create manifest (already merged in)
- [x] Boot image and system image build successfully
- [x] Device boots into rootfs, `usb: Manufacturer: GNU/Linux Device` appears in `dmesg` on host.
- [x] LXC container starts and does not crash
- [ ] libhybris tests
  * [ ] test_gps
  * [x] test_hwcomposer (kinda)
  * [x] test_lights
  * [x] test_vibrator
  * [ ] test_wifi
  * [ ] test_sensors
  * [ ] test_audio
  * [ ] test_camera
  * [ ] test_input
  * [ ] test_nfc
  * [ ] test_recorder 
  * [x] test_egl_configs

I'm assuming hwcomposer works, but since I cannot connect to my WiFi in my house, I cannot upgrade to the caf repo and test. test_hwcomposer changes the screen to a black screen from a boot screen though.

**Comment by LNJ2** [comment]
I don't think vold is required for halium, don't worry about vold.

**Comment by ReubenBTalbott** [comment]
How could I download or build this?

**Comment by Qiangong2** [comment]
@ReubenBTalbott just sync the sources and download the jfltetmo trees, it should be already included in the repos

**Comment by ReubenBTalbott** [comment]
Are the T-Mobile and US cellular Samsungs S4's identical enough that I can use this on a US cellular S4?

**Comment by Qiangong2** [comment]
Possibly, since there are no hardware differences except for the modem (to my knowledge). The build should work if you just replace the jfltetmo tree with a jflteusc tree. It would need to be cm-14.1 though. cm-14.0 will not work.

**Comment by ReubenBTalbott** [comment]
Thank you.

**Comment by ReubenBTalbott** [comment]
I am getting a little confused on how to build this, are there some pre built files I can flash?

**Comment by Qiangong2** [comment]
Not at the moment, I'll see if I can get some working

**Comment by ReubenBTalbott** [comment]
Thanks a lot.

**Comment by Qiangong2** [comment]
I should get the image up within the next day or so

**Comment by ReubenBTalbott** [comment]
Would I flash it with TWRP?
Also thanks for doing this.

**Comment by Qiangong2** [comment]
No you wouldn't be able to flash with twrp, you would have to use JBB Gameich's halium install script on a linux machine or VM, You would also need Heimdall (or Odin) to flash the hybris-boot image. Refer to the halium docs for install guide

**Comment by ReubenBTalbott** [comment]
This?
https://docs.halium.org/en/latest/porting/install-build/reference-rootfs.html

**Comment by Qiangong2** [comment]
Yes, However, use this script instead: https://github.com/JBBgameich/halium-install

**Comment by Qiangong2** [comment]
@ReubenBTalbott 

Here are the hybris-boot image and system image. It works with Plasma Mobile only. There is no guarantee these will work with any other device than the jfltetmo as I do not have access to any others for testing.

As always, if you brick your phone, it isn't my fault :stuck_out_tongue: 

Note: You will have to generate the udev rules inside of the image yourself
https://www.androidfilehost.com/?fid=674106145207488733
https://www.androidfilehost.com/?fid=818222786056032275

**Comment by ReubenBTalbott** [comment]
Is this the rootfs i want? https://www.plasma-mobile.org/get/

**Comment by ReubenBTalbott** [comment]
I get the same issue with the rootfs  from halium and the one from Plasma
Mobile, i can detect the usb network device but i can't ssh it, I have
tried using a ubuntu 18.04 laptop and a iMac running high sierra. This is
the error: root@DevBook:/home/reuben# ssh: connect to host 10.15.19.82 port
22: No route to host

Do you know what I am doing wrong?

Thanks

**Comment by Qiangong2** [comment]
You haven't set the ip to 10.15.19.100 like the docs say. There is info on it on the web if you search.

Also, make sure you are using the CAF rootfs.

**Comment by ReubenBTalbott** [comment]
Ok i'll try that, thanks

**Comment by ReubenBTalbott** [comment]
Does this look right?
![screen shot 2018-05-23 at 4 11 45 pm](https://user-images.githubusercontent.com/19376692/40448640-1f6518a8-5ea4-11e8-93fa-195c46a3166f.png)

**Comment by ReubenBTalbott** [comment]
opps wrong ip
![screen shot 2018-05-23 at 4 13 08 pm](https://user-images.githubusercontent.com/19376692/40448672-327da810-5ea4-11e8-9300-e311178195df.png)

**Comment by Qiangong2** [comment]
Yes, although I don't know how it works on Macs, I usually change it via the terminal on Linux, but that seems correct.

**Comment by ReubenBTalbott** [comment]
It was not working on mac, what I was doing seems to have been setting the
macs ip, could you tell me the command for linux? I have looked around for
a bit and can't find the right command (everything that i find is for
changing the computers ip.)

**Comment by Qiangong2** [comment]
https://danielmiessler.com/study/set_ip/

Just change the name of the interface to match

**Comment by ReubenBTalbott** [comment]
Does the plasma desktop work yet?

**Comment by Qiangong2** [comment]
You mean Plasma Mobile?

**Comment by ReubenBTalbott** [comment]
Yes, as in does the plasma mobile desktop GUI work on the phone.

**Comment by JBBgameich** [comment]
Before the gui can load, you need to generate the udev rule. Otherwise it has no permission to draw to the screen.

**Comment by ReubenBTalbott** [comment]
Okay thanks, do I follow the halium docs for generating the udev rule?

**Comment by JBBgameich** [comment]
Exactly. Please pull-request the result in the lxc-android repository in case it's not yet there.

**Comment by ReubenBTalbott** [comment]
I'll try, but I have never done a pull-request before, so we will see how it goes.

**Comment by Qiangong2** [comment]
@JBBgameich Even though I have the jfltetmo udev rules in the halium lxc-android repo, it doesn't copy them over in the build.

**Comment by JBBgameich** [comment]
lxc-android included in the rootfs. It just needs to get rebuilt on the ci.

**Comment by Qiangong2** [comment]
Even the latest one doesn't have it. And I merged it in a couple moths ago

**Comment by bhush9** [comment]
This should be fixed in next rootfs build upcoming week.

**Comment by ReubenBTalbott** [comment]
Okay i got ssh access and about to create the udev rules, which code name should i use? the T mobile one or the US cellular one?

**Comment by ReubenBTalbott** [comment]
How do i get the udev rules? are they in the lxc-android repo, or do i follow the [halium docs](https://docs.halium.org/en/latest/porting/debug-build/udev.html)?

**Comment by JBBgameich** [comment]
You can use the ones from the lxc-android repository (https://raw.githubusercontent.com/Halium/lxc-android/master/usr/lib/lxc-android/70-jfltetmo.rules), but generating them yourself will bring the same result.

**Comment by ReubenBTalbott** [comment]
Do i delete the other .rules files?
![screen shot 2018-05-26 at 11 58 04 am](https://user-images.githubusercontent.com/19376692/40578103-2135e36c-60dc-11e8-89d3-c0539d6858dc.png)

**Comment by JBBgameich** [comment]
No, that's not neccesary.

**Comment by ReubenBTalbott** [comment]
What do i do now?

**Comment by ReubenBTalbott** [comment]
How do i start the plasma desktop?

**Comment by JBBgameich** [comment]
It should start automatically if you are using the correct rootfs and the udev rules are in place.

**Comment by ReubenBTalbott** [comment]
ok, how do i increase space apt has so it can update?
![screen shot 2018-05-26 at 12 06 07 pm](https://user-images.githubusercontent.com/19376692/40578148-309357d0-60dd-11e8-9213-b919e029be15.png)

**Comment by ReubenBTalbott** [comment]
Its stuck at the samsung logo, i used the latest Plasma mobile CAF rootfs, and I created the udev, do i have to activate the udev rules?

**Comment by JBBgameich** [comment]
You need to reboot once. I just re-read the conversation here and since you are not using the exact same device, maybe there's a difference between them which makes the gui not start. You could try rebuilding the image using the device tree for this device.

**Comment by ReubenBTalbott** [comment]
I have looked and this device is supposed to be compatible, I think i did something wrong. are there any commands i can run to look at logs?

**Comment by JBBgameich** [comment]
No, I don't think so. It's just that the same specifications don't mean that the device is compatible. There are a few things you could check:
* is there a vendor partition on the device?
* if yes, which android version was installed before?
* Does the lxc container really start? `systemctl status lxc@android`

**Comment by ReubenBTalbott** [comment]
1 not sure i will check,
2 i had lineage os 14.1 which is based on android 7, 
3 checking now.
Thanks for the reply

**Comment by ReubenBTalbott** [comment]
This is the log of the lxc container 
![screen shot 2018-05-26 at 12 34 31 pm](https://user-images.githubusercontent.com/19376692/40578382-35771210-60e1-11e8-9a97-5f2b94a79e25.png)

**Comment by ReubenBTalbott** [comment]
And this is what i get if i try to manually 
![screen shot 2018-05-26 at 12 39 35 pm](https://user-images.githubusercontent.com/19376692/40578432-f0274e90-60e1-11e8-8de8-3b5390bcdb48.png)
start it

**Comment by JBBgameich** [comment]
Please never start the container manually, it might delete content on the data partition unintentionally.

**Comment by JBBgameich** [comment]
The non-starting lxc container is a fundamental problem which would explain the issues. @Qiangong2 It did already start on your device I think, can you check if it really still does with the build @ReubenBTalbott is using?

**Comment by Qiangong2** [comment]
@JBBgameich it works fine for me, I don't know why it isn't working for @ReubenBTalbott. However, Sailfish OS for jfltexx doesn't work on the jfltetmo even though android ROMs are compatible. Maybe something similar is happening here?

**Comment by dkati** [comment]
hello guys.jfltexx lineageOS dev here.
Do you actually managed to build systemimage without any edits on /build/core/main.mk ?

**Comment by Qiangong2** [comment]
@dkati yes. The only thing I changed was the defconfig. However, I don't have a jfltexx to test. Only a jfltetmo

**Comment by dkati** [comment]
can i help ? do we have a buildable manifest ? @Qiangong2

**Comment by jbruechert** [comment]
We collect the manifests in the halium-devices repo. The one for this device is here: https://github.com/Halium/halium-devices/blob/halium-7.1/manifests/samsung_jfltetmo.xml

In a halium tree you can use the setup script included in the repository to download the sources and run some automatic modifications.

**Comment by YamiYukiSenpai** [comment]
When I ran `mka mkbootimg` for jfltexx, I'm getting
```
halium/hybris-boot/Android.mk:69: ********************* /boot appears to live on /dev/block/platform/msm_sdcc.1/by-name/boot
halium/hybris-boot/Android.mk:70: ********************* /data appears to live on /dev/block/platform/msm_sdcc.1/by-name/userdata
```

Pretty sure the mountpoints should be fixed, or am I missing anything?

Here's the [full output](https://pastebin.com/uTBSnuv8)

**Comment by Qiangong2** [comment]
@YamiYukiSenpai Again, I only have a jfltetmo. I can only speak for at device. It looks like (from the error) that the partitions haven't been added to fixup-mountpoints yet

**Comment by YamiYukiSenpai** [comment]
I know you have a jfltetmo, but I figured I could get some insight from you.

I double checked the fixup-mountpoints before I ran `mka mkbootimg` when that error first showed up, I copied the mountpoints from jfltetmo. Still the same error.

List of commands I ran to build halium:
* `source build/envsetup.sh`
* `breakfast jfltexx`
* `mer-kernel-check/mer_verify_kernel_config kernel/samsung/jf/arch/arm/configs/lineageos_jf_defconfig`
* `mka mkbootimg`

I'm posting the output for each commands [here](https://pastebin.com/hzR87yD0).  Thanks for your time.

**Comment by Qiangong2** [comment]
@YamiYukiSenpai You have to download the SettingsLib repo from lineageOS for it to start building. I don't know why, but iof you don't have that, it will not build Halium properly.

**Comment by jbruechert** [comment]
Components requiring SettingsLib should automatically be removed by halium-devices nowadays. If not, it's recommended to patch the device tree instead of including settingsLib.

**Comment by Qiangong2** [comment]
@JBBgameich I haven't built it recently, but a few months ago, the jfltetmo version wouldn't even boot without me including the SettingsLib repo

**Comment by YamiYukiSenpai** [comment]
Is it [this one](https://github.com/LineageOS/android_packages_apps_Settings/tree/cm-14.1)?

**Comment by Qiangong2** [comment]
You need [this folder](https://github.com/LineageOS/android_frameworks_base/tree/cm-14.1/packages/SettingsLib)

and put it in frameworks/base/packages

**Comment by YamiYukiSenpai** [comment]
Now I'm getting [this](https://pastebin.com/sQXkyjic) when I ran `mka hybris-boot`:
```
ninja: build stopped: subcommand failed.
build/core/ninja.mk:151: recipe for target 'ninja_wrapper' failed
make: *** [ninja_wrapper] Error 1
make: Leaving directory '/home/yamiyukisenpai/Projects/jfltexx_halium/halium'

#### make failed to build some targets (8 seconds) ####
```

**Comment by Qiangong2** [comment]
Did you do `mka clean` between builds? Did you remember to do `make mkbootimg`?

**Comment by YamiYukiSenpai** [comment]
@Qiangong2 sorry for replying so late.  I just recently got back to it

And yes, I did use `mka clean` in between.

https://pastebin.com/NQc3ViJg

**Comment by Qiangong2** [comment]
Try using `export USE_NINJA=false` before `source build/envsetup.sh`

**Comment by Qiangong2** [comment]
Due to the fact that the S4 uses a 3.4 kernel, I am unsure if it is able to run Halium anymore. If anyone can confirm that would great. That I think is why it hasn't boot since November 2017

**Comment by Qiangong2** [comment]
I got it working! Plasma now boots again :D Thanks Herrie! Now to iron out the kinks and then try ubports

**Comment by YamiYukiSenpai** [comment]
How do you fix this?  I tried following [this section in documented errors](http://docs.halium.org/en/latest/porting/common-system-build-errors.html#hybris-boot-part-and-hybris-data-part),  but it didn't help, and it seems to be a bit different from what I'm getting.

```
Checking build tools versions...
frameworks/av/camera/cameraserver/Android.mk:18: Target has integrated cameraserver into mediaserver. This is weakening security measures introduced in 7.0
halium/hybris-boot/Android.mk:69: ********************* /boot appears to live on /dev/block/platform/msm_sdcc.1/by-name/boot /dev/mmcblk0p20
halium/hybris-boot/Android.mk:70: ********************* /data appears to live on /dev/block/platform/msm_sdcc.1/by-name/userdata /dev/mmcblk0p29
halium/hybris-boot/Android.mk:73: *** There should be a one and only one device entry for HYBRIS_BOOT_PART and HYBRIS_DATA_PART.  Stop.
```

**Comment by Qiangong2** [comment]
@YamiYukiSenpai the fstab files should say the exact same as the fixup mountpoints. It looks like you have two differing ones in your tree. Check out my jf-common tree and the jfltetmo portion of fixup-mountpoints

**Comment by YamiYukiSenpai** [comment]
I ran `fixup-mountpoints` just before I ran `mka mkbootimg`.  Double checked them all to make sure every partition entry in `<src>` column was changed.  I also tried copying jfltetmo's entry in fixup-mountpoint, but that didn't help either.

**Comment by Qiangong2** [comment]
And you're positive that what is listed in fstab.qcom and twrp.fstab are the same as what is in fixup-mountpoints?

**Comment by YamiYukiSenpai** [comment]
yes

**Comment by dkati** [comment]
Short update. Using ubuntu touch rootfs, i managed to reach the boot animation screen...i cant connect with ssh "connection refused" so i am stuck here. I have a complete manifest you just sync and build.

I build for unified jflte
